### ISSUE TITLE
Debugging broken after commit 2ba639c99460a9f7272da0959cdcfa948bd65ded

### ISSUE THREAD

**Issue by loeiten** [boutproject/BOUT-dev#176]
Debugging after commit 2ba639c99460a9f7272da0959cdcfa948bd65ded is broken.

At 2ba639c99460a9f7272da0959cdcfa948bd65ded,
```
./configure --with-checks=3 --with-track --with-debug --with-petsc
```
results in
```
----- Compiling BOUT++ -----
CXX      =  mpicxx
CFLAGS   =  -g -O0 -DCHECK=3 -DSIGHANDLE -DTRACK -DREVISION=2ba639c99460a9f7272da0959cdcfa948bd65ded -DNCDF -DBOUT_HAS_PETSC -DBOUT_HAS_PETSC_DEV -DBOUT_HAS_PVODE
```
when making, whilst the current master yields

```
----- Compiling BOUT++ -----
CXX      =  mpicxx
CFLAGS   =  -g -O2 -g -O0 -Og -DCHECK=3 -DSIGHANDLE -DTRACK -DREVISION=872971cc24a73e2640406326680013f2ca11cbac -DNCDF -DBOUT_HAS_PETSC -DBOUT_HAS_PETSC_DEV -DBOUT_HAS_PVODE
```

i.e. there are some spurious `-g -O2` optimization flags, which makes the debugger jumpy, as was the case in #30

**Comment by dschwoerer** [comment]
What compiler are you using?
gcc should only care about the last -O? flag, and ignore the previous ones:

> If you use multiple -O options, with or without level numbers, the last such option is the one that is effective.

https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

**Comment by loeiten** [comment]
I see, thanks.

The debugging is still jumpy with the new configure though. Could it be the `-Og` optimization need to preceded with a `-g` flag?

I'm by the way using

```
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.1) 4.8.4
```

**Comment by dschwoerer** [comment]
Could you specify what you mean with jumpy - are instruction skipped?
Do you have a minimal example to reproduce this? I assume you are using gdb?

Could you test if it is related to the `-Og` flag? Maybe this flag already optimizes to much and we should switch back to the `-O0`.

**Comment by loeiten** [comment]
By jumpy I mean that `next` in gdb is not going to the next instruction. By consecutively pressing `next` gdb will pass through a function in an almost arbitrary manner.

I had a minimal example, which I deleted by accident. Before making a new minimal example I cleaned my local repo folder by deleting everything, and cloning again. After trying the new minimal example, it looks like I am not able to reproduce this bug.

I really do not know what happen the last time, but I will close this issue, and eventually reopen it if I find that the bug reoccurs.

Thanks for the help anyway :).
